torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -666,17 +668,15 @@ def view_avoid_dupes_with_primals(t):
         )
         user_outs = pytree.tree_map(from_fun, f_output_tangents)
 
-        if (
-            torch._dynamo.config.inline_inbuilt_nn_modules
-            or torch._dynamo.compiled_autograd.in_compiled_autograd_region
-        ):
-            static_parameter_input_indices = [
+        nonlocal static_input_indices
+        static_input_indices = static_input_indices or []
+        if torch._dynamo.compiled_autograd.in_compiled_autograd_region:
+            passed_indices = set(static_input_indices)
+            static_input_indices = [
                 i
                 for i, arg in enumerate(flat_args)
-                if isinstance(arg, torch.nn.Parameter)
+                if (isinstance(arg, torch.nn.Parameter) or i in passed_indices)
```

Comment:
hmm how is it possible to have a `torch.nn.Parameter` arg here that is not in `static_input_indices` already? I thought dynamo would populate that list with all indices that correspond to params/buffers already.

Reply:
Oh mayeb one answer is if we're no longer gating this logic on `config.inline_inbuilt_nn_modules` (above), then the params/buffers will not show up as placeholders in the `GraphModule` passed in from dynamo so you need to manually add their indices here.

Is that right? I would probably branch the logic on `if torch._dynamo.config.inline_inbuilt_nn_modules` (to make it clear that when that config is true, we don't actually need to do any work here to find extra indices)

Reply:
This part was actually interesting - dynamo should already mark these, but in the compiled autograd case, I think it creates some that aren't marked by dynamo (there was one  specific test which relied on this behavior.

Reply:
It turns out after some investigation that due to the different calling convention utilized in compiled autograd that the index metadata is missed, I can move the carveout from AOT to dynamo for this case and read the metadata off of the list input. (separate PR)